src/pkg/bundler/localbundle.go

```diff
@@ -113,29 +113,45 @@ func (lo *LocalBundle) create(signature []byte) error {
 	// push bundle manifest to OCI stoer
 	var bundleManifestDesc ocispec.Descriptor
 	if lo.isTofu {
-		bundleManifestDesc, err = pushBundleTFToStore(store, filepath.Join(lo.sourceDir, config.BundleTF))
+		bundleManifestDesc, err = pushFileToStore(store, filepath.Join(lo.sourceDir, config.BundleTF), config.BundleTF)
 	} else {
 		bundleManifestDesc, err = pushBundleYAMLToStore(store, bundle)
 	}
 	if err != nil {
 		return err
 	}
 
-	// append uds-bundle.yaml layer to rootManifest and grab path for archiving
+	// append uds-bundle.yaml (or uds-bundle.tf) layer to rootManifest and grab path for archiving
 	rootManifest.Layers = append(rootManifest.Layers, bundleManifestDesc)
 	digest := bundleManifestDesc.Digest.Encoded()
 	artifactPathMap[filepath.Join(lo.tmpDstDir, config.BlobsDir, digest)] = filepath.Join(config.BlobsDir, digest)
 
+	// if source is a .tf file, push an additional config file and the uds_provider to the bundle
 	if lo.isTofu {
-		bundleConfigManifestDesc, err := pushBundleTFConfigToStore(store, bundle)
+		// append uds-tf-config.yaml layer to rootManifest and grab path for archiving
+		tfConfig := types.TFConfigHelper{Packages: bundle.Packages}
+		bundleConfigManifestDesc, err := pushBundleTFConfigToStore(store, tfConfig)
 		if err != nil {
 			return err
 		}
-
-		// append uds-bundle.yaml layer to rootManifest and grab path for archiving
 		rootManifest.Layers = append(rootManifest.Layers, bundleConfigManifestDesc)
 		digest := bundleConfigManifestDesc.Digest.Encoded()
 		artifactPathMap[filepath.Join(lo.tmpDstDir, config.BlobsDir, digest)] = filepath.Join(config.BlobsDir, digest)
+
+		// append the terraform provider binary to the bundle
+		// TODO:: This is pulling from the local filesystem until we start publishing the provider externally
+		goBinPath := os.Getenv("GOBIN")
+		if goBinPath == "" {
+			return fmt.Errorf("GOBIN environment variable was not set, unable to locate the locally installed terraform-provuder-uds binary")
+		}
+		providerPath := filepath.Join(goBinPath, config.TerraformProvider)
```

Comment:
We may want to pull this from the `.terraform` directory that gets created with `tofu init` instead. It's probably normal to not have the provider actually installed in the gobin and instead just pulled down in to the CWD `.terraform`. We could probably use `filepath.Walk()` to find it. 

- https://pkg.go.dev/path/filepath#Walk

For the sake of this PR we could leave it as-is though and make it a follow-up issue. Up to you.

Reply:
Why do today what you can do tomorrow?

> It's probably normal to not have the provider actually installed in the gobin and instead just pulled down in to the CWD.

I agree that looking in GOBIN is not a good solution, but we should follow up with the create flow to make sure we're on the same page. I think I'm operating off the mindset that the things `tofu init` would do should be handled automagically when people use the UDS CLI?

Reply:
I agree `init` should likely have been called already, but `init` does not install the provider in `GOBIN` path, if that's what you're thinking?